### DISCUSSION TITLE
Aspect ration not working

### DISCUSSION THREAD

**Discussion by whatnowangga** [shogo4405/HaishinKit.dart#6] [Q&A]
Hi
i already try for ios and working properly as i expected
but seems video configuration not work so well, 
i set to 1080x1920
but the return is still landscape with 480x272

just saw the CaptureSettings is dont have any entity,
can someone help me?

**Comment by Kodam-zz** [comment]
I created a pull request( #7 ) to fix this issue

**Reply by whatnowangga** [reply]
Seems still not work, 
i already try with your pull request code
seems videoSetting mean is for player video,
and i check the capture setting is dont have any entity
any thought?

**Comment by Kodam-zz** [comment]
Sorry to hear that. Did you try it on Android or iOS?

It does work for me if I set the video settings before the calling the
connect method. Changing the width/height later doesn't have any effect

**Reply by whatnowangga** [reply]
i try in iOS, 
android work well without any issue

**Comment by whatnowangga** [comment]
would you mind to share your code for publish rtmp? @Kodam-zz

**Reply by Kodam-zz** [reply]
Are you sure you used the correct version? Sometimes pub doesn't download the new version of the lib if a git address is set. Remove the pubspec.lock and clean the project.

Here is the code I used in the project (_stream is a ValueNotifier ):

``` 
...
      RtmpConnection connection = await RtmpConnection.create();
      connection.eventChannel.receiveBroadcastStream().listen((event) {
        print(event);
        switch (event["data"]["code"]) {
          case 'NetConnection.Connect.Success':
            if (streamId.value == null) return;
            _stream.value?.publish(streamId.value!);
            _recording.value = true;
            break;
        }
      });
      RtmpStream stream = await RtmpStream.create(connection);
      stream.attachAudio(AudioSource());
      stream.captureSettings = CaptureSettings();
      stream.attachVideo(VideoSource(position: currentPosition.value));
      stream.videoSettings = VideoSettings(
        width: 720,
        height: 1280,
        bitrate: 2000000,
        frameInterval: 2,
        muted: false,
      );
    });
....
_connection.value?.connect("rtmp://global-live.mux.com:5222/app/");
....
```

**Comment by whatnowangga** [comment]
![image](https://user-images.githubusercontent.com/13533060/194541862-0c919bae-bad4-44fa-8501-9667d4dc2a94.png)

i follow your video settings, but seems width and height not as expected

**Comment by whatnowangga** [comment]
im using the latest release, and getting the package by git, im pretty sure im using the latest

**Comment by AlexUrrutia** [comment]
This is happening to me as well...
see the third stream showing 0x0 in the resolution ...
Running on iOS

<img width="625" alt="Screen Shot 2023-06-03 at 11 39 59 PM" src="https://github.com/shogo4405/HaishinKit.dart/assets/7042361/66df15e6-f18a-4ac7-94e0-4c2fc88a83d5">

**Comment by hardikthakkar2208** [comment]
Hello, Any solution for the above issue? i am facing same issue
